packages/hash/frontend/src/components/layout/PageHeader/SearchBar/index.tsx

```diff
@@ -168,21 +177,58 @@ const SearchBarWhenSearchIsEnabled: React.VFC = () => {
       }}
       ref={rootRef}
     >
-      {!isMobile ? (
-        <DesktopSearch
-          displayedQuery={displayedQuery}
-          setQueryText={setQueryText}
-          setResultListVisible={setResultListVisible}
-        />
+      {/* If the user is in mobile view and the search icon isn't clicked, display the icon */}
+      {isMobile && !displaySearchInput ? (
+        <IconButton
+          sx={{
+            height: "100%",
+            display: "flex",
+            alignItems: "center",
+            mr: 2,
+          }}
+          onClick={() => setDisplaySearchInput(true)}
+        >
+          <SearchIcon sx={{ height: theme.spacing(2), width: "auto" }} />
+        </IconButton>
       ) : (
-        <MobileSearch
-          displayedQuery={displayedQuery}
-          setQueryText={setQueryText}
-          setResultListVisible={setResultListVisible}
-          displaySearchInput={displaySearchInput}
-          setDisplaySearchInput={setDisplaySearchInput}
-        />
+        <Box
+          style={{
+            display: "flex",
+            width: "100%",
+            background: "white",
+            zIndex: 1,
+          }}
+        >
+          <SearchInput
+            displayedQuery={displayedQuery}
+            isMobile={isMobile}
+            setQueryText={setQueryText}
+            setResultListVisible={setResultListVisible}
+          />
+
+          {isMobile && (
+            <Box
+              component="span"
+              sx={{
+                marginLeft: theme.spacing(2),
+                display: "flex",
+                alignItems: "center",
+              }}
+            >
+              <Button
+                onClick={() => {
+                  setQueryText("");
+                  setDisplaySearchInput(false);
+                }}
+                variant="tertiary_quiet"
+              >
+                Cancel
+              </Button>
+            </Box>
+          )}
```

Comment:
Non-blocking, but I don't see why this button needs to be wrapped into a `span`. If it's to center the button in the flex row, consider adding `alignItems: center` to the Box component on line 194.